.github/workflows/esp32_build.yml

```diff
@@ -0,0 +1,89 @@
+name: ESP32 Build
+
+on: [push, pull_request, workflow_dispatch]
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false  # don't cancel if a job from the matrix fails
+      matrix:
+        config: [
+            esp32buzz,
+        ]
+        gcc: [10]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - name: Install Prerequisites
+        shell: bash
+        run: |
+          sudo apt-get install git wget libncurses-dev flex bison gperf python3 python3-pip python3-setuptools python3-serial python3-cryptography python3-future python3-pyparsing python3-pyelftools cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
+          sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+          rm -rf /usr/local/bin/cmake
+
+          sudo apt remove --purge --auto-remove cmake
+          sudo apt update && \
+          sudo apt install -y software-properties-common lsb-release && \
+          sudo apt clean all
+          wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
```

Comment:
why using this Cmake version ? 
thus it doesn't match kiware instructions : https://apt.kitware.com/ ?

Reply:
cmake-upgrade taken verbatim from the top answer here: https://askubuntu.com/questions/355565/how-do-i-install-the-latest-version-of-cmake-from-the-command-line